src/gengowatcher/ui_textual.py

```
@@ -0,0 +1,1124 @@
+
+    @work(thread=True)
+    def refresh_worker(self) -> None:
+        """Background worker to refresh UI from watcher state."""
+        while not self.watcher.shutdown_event.is_set():
+            try:
+                # Check if app is still running before calling
+                if self._exit:
+                    break
+                self.call_from_thread(self._refresh_ui)
+                self.call_from_thread(self._drain_log_queue)
+            except Exception:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)